components/lwip/port/esp32/include/lwipopts.h

```diff
@@ -511,6 +511,49 @@
 #endif
 
 
+#endif
+
+/*
+   ------------------------------------
+   ---------- 6LoWPAN options ----------
+   ------------------------------------
+*/
+
+#ifdef CONFIG_LWIP_LOWPAN6_SUPPORT
+
+/**
+ * LOWPAN6_DEBUG: Enable debugging for 6LoWPAN.
+ */
+#define LWIP_LOWPAN6_DEBUG_ON           CONFIG_LWIP_LOWPAN6_DEBUG_ON
+
+#if LWIP_LOWPAN6_DEBUG_ON
```

Comment:
```suggestion
#ifdef CONFIG_LWIP_LOWPAN6_DEBUG_ON
```

Nitpick: Just to shorten, simplify the header.
This works in IDF (we don't compile with `-Wundef`), but might cause trouble when build using lwip flags, where this could be treated as error. (Note that Kconfig macros are either defined as `1` (`yes`) or undefined (`'no`))

The same comment for the below config options.

Reply:
I was actually just directly coping the style of SLIPIF above, thinking it was needed in some unknown way.
However after looking over the rest of the header, doing it with ifdef is much neater.

Reply:
Yes, we should cleanup the `lwipiopt.h` and use one common way of converting the Kconfig options to lwip defines. Thanks for the update.

> 
> The LWIP_LOWPAN6_DECOMPRESSION_DEBUG & LWIP_RFC7668_IP_UNCOMPRESSED_DEBUG options are very useful. In fact I was using them last week.
> But they are VERY verbose, having them enabled when you simply need the minimal output from LWIP_LOWPAN6_DEBUG would get distracting (and annoying!).

Okay, then it makes sense to keep them configurable separately.

---

Thanks again for your contribution. will move this PR to the internal round our reviews and testing, but please be patient, this will take some time.